apps/src/templates/progress/progressHelpers.js

```diff
@@ -194,3 +206,87 @@ export const processedLevel = level => {
       level.sublevels && level.sublevels.map(level => processedLevel(level))
   };
 };
+
+const getLevelResult = serverProgress => {
+  if (serverProgress.status === LevelStatus.locked) {
+    return TestResults.LOCKED_RESULT;
+  }
+  if (serverProgress.readonly_answers) {
+    return TestResults.READONLY_SUBMISSION_RESULT;
+  }
+  if (serverProgress.submitted) {
+    return TestResults.SUBMITTED_RESULT;
+  }
+
+  return serverProgress.result || resultFromStatus(serverProgress.status);
+};
+
+/**
+ * Parse a level progress object that we get from the server using either
+ * /api/user_progress or /dashboardapi/section_level_progress into our
+ * canonical studentLevelProgressType shape.
+ * @param {object} serverObject A progress object from the server
+ * @returns {studentLevelProgressType} Our canonical progress shape
+ */
+export const levelProgressFromServer = serverObject => {
+  return {
+    status: serverObject.status || LevelStatus.not_tried,
+    result: getLevelResult(serverObject),
+    paired: serverObject.paired || false,
+    timeSpent: serverObject.time_spent || 0,
+    pages:
+      serverObject.pages_completed && serverObject.pages_completed.length > 1
```

Comment:
Should this be `serverObject.pages_completed.length > 0`, should there be mapping if there is 1 `pages_complete`?

Reply:
nope. "pages" is only used by multi-page assessments, and its presence (or absence) is use to indicate that level type. i will add a comment explaining that!